src/server/page_server.cpp

```diff
@@ -995,6 +996,9 @@ page_server::execute_catchup (cubthread::entry &entry, const LOG_LSA catchup_lsa
   size_t request_page_cnt = std::min (log_pgptr_recv_vec.size (), remaining_page_cnt);
   auto req_future = std::async (std::launch::async, request_pages_to_buffer, request_start_pageid, request_page_cnt);
   int error = NO_ERROR;
+
+  LOG_CS_ENTER (&entry);
```

Comment:
I may not remember it well... 
but what does this critical section protect against?

Reply:
Honestly, I don't know exact lock scope of this critical section, but it's acquired when modifying some variables I modify in this operation and calling some function I used.
For example, ` log_Gl.hdr.append_lsa` and `logpb_flush_all_append_pages`. 

Additionally, I considered reducing the critical section by holding the locking only while appending, not receiving from the followee. But, I doubt it works because the state of the log_manager is not concluded while appending pages. For example, before the `logpb_catchup_finish()` which is the last call of the catch-up process, the `append_lsa` points to a middle of log records. If another thread flushes pages at this point, it will append EOF log record in the middle of log. It would happen if we release the lock halfway through the process.

Reply:
I apologize for the unclear question. I agree that LOG_CS_ENTER() is not very clear and appears to protect too many objects. Every time changes are made to the log buffer, log volume, or log_Gl, it enters a critical section.

I initially thought that replication wouldn't proceed while PS was catching up. I believed that only catch-up threads had access to the log buffer, log volume, and log_Gl. Therefore, I questioned the need to enter the critical section. However, if replication also occurs simultaneously, it seems appropriate to enter the critical section through LOG_CS_ENTER(). This is because replication involves accessing the log buffer/volume and obtaining pages while it proceeds.

Reply:
No need to apologize. Actually, it brings up an issue I missed. It seemed that the replicator works well along with the catch-up mechanism in parallel without any modification. And, it did when I tested, but there are some cases it can't. The replicator conditionally acquires the lock such as fetching a log page from disk. In this case, the replicator gets stuck and waits for the catch-up to be done. I have to address this issue to make them work in parallel.

Regarding when to deal with this, this issue was made earlier as a TODO: http://jira.cubrid.org/browse/LETS-753. I was going to just close this, but it still has points. Let me address the problem in the issue.